src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyCallSiteSelector.java

```diff
@@ -43,7 +44,27 @@
  */
 class GroovyCallSiteSelector {
 
-    private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Object[] parameters) {
+    private static boolean matches(@Nonnull Class<?>[] parameterTypes, @Nonnull Object[] parameters, boolean varargs) {
+        if (varargs) {
+            int fixedLen = parameterTypes.length - 1;
+            Class<?> componentType = parameterTypes[fixedLen].getComponentType();
+            assert componentType != null;
+            int arrayLength = parameters.length - fixedLen;
+            if (arrayLength >= 0) {
+                if (arrayLength == 1 && parameterTypes[fixedLen].isInstance(parameters[fixedLen])) {
+                    // not a varargs call
+                } else {
+                    Object array = Array.newInstance(componentType, arrayLength);
+                    for (int i = 0; i < arrayLength; i++) {
+                        Array.set(array, i, parameters[fixedLen + i]);
```

Comment:
> then I think this could be done more simply with one of the `Arrays.copyOfRange` methods?

No, these are for arrays whose component type is statically known.